### DISCUSSION TITLE
Crash when running example missing gstreamer plugin

### DISCUSSION THREAD

**Discussion by FerroEduardo** [TeleMidia/ginga#179] [General]
System: `Ubuntu 20.04.6` on WSL 2
Kernel: `5.15.90.1-microsoft-standard-WSL2`
Script: `primeiro-joao/01sync.ncl`

Terminal
```
$ ginga examples/primeiro-joao/01sync.ncl
**
ERROR:PlayerVideo.cpp:95:ginga::PlayerVideo::PlayerVideo(ginga::Formatter*, ginga::Media*): '_audio.pan' should not be nullptr
Bail out! ERROR:PlayerVideo.cpp:95:ginga::PlayerVideo::PlayerVideo(ginga::Formatter*, ginga::Media*): '_audio.pan' should not be nullptr
Aborted
```
------------------
I don't know if this is related to WSL, but I was able to run others scripts:
Pacman
![image](https://github.com/TeleMidia/ginga/assets/47820549/7107b82f-f8aa-4916-a669-b19918608eab)

Luarocks
![image](https://github.com/TeleMidia/ginga/assets/47820549/a37de044-d4f7-40be-a4c9-44b1381f6cf6)

**Comment by alanlivio** [comment]
Many thanks @FerroEduardo. Did you have instaled the gstreamer1.0-plugins-bad package in WSL (see [here](https://github.com/TeleMidia/ginga/tree/master/extra/ubuntu))?
Some decoders are dynamic loaded by gstreamer.

**Comment by FerroEduardo** [comment]
> Many thanks @FerroEduardo. Did you have instaled the gstreamer1.0-plugins-bad package in WSL (see [here](https://github.com/TeleMidia/ginga/tree/master/extra/ubuntu))? Some decoders are dynamic loaded by gstreamer.

After installing `gstreamer` packages, I was able to run with success!
Thanks a lot! :)
![image](https://github.com/TeleMidia/ginga/assets/47820549/c6dc908a-aefd-4d4d-ac24-922435fd4a0e)

If helps somehow, I also got this output
```
$ ginga ginga/examples/primeiro-joao/01sync.ncl
Cannot connect to server socket err = No such file or directory
Cannot connect to server request channel
jack server is not running or cannot be started
JackShmReadWritePtr::~JackShmReadWritePtr - Init not done for -1, skipping unlock
JackShmReadWritePtr::~JackShmReadWritePtr - Init not done for -1, skipping unlock
Cannot connect to server socket err = No such file or directory
Cannot connect to server request channel
jack server is not running or cannot be started
JackShmReadWritePtr::~JackShmReadWritePtr - Init not done for -1, skipping unlock
JackShmReadWritePtr::~JackShmReadWritePtr - Init not done for -1, skipping unlock
```

**Reply by alanlivio** [reply]
This JackShmReadWritePtr is probably a lock issue in the PuselAudio server of WSLg.

**Comment by rafael2k** [comment]
This is related to Jack, not PA, and can be safely ignored.
